enumerable.rb

```diff
@@ -0,0 +1,199 @@
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  ## method to check incoming data type
+  def check_data_type(data_type)
+    arr = data_type if data_type.instance_of?(Array)
+    arr = to_a if data_type.instance_of?(Range)
+    arr = data_type if data_type.instance_of?(Hash)
+    arr
+  end
+
+  ## my_each method
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = check_data_type(self)
+    for i in arr
+      yield(i)
+    end
+    arr
+  end
+
+  ## my_each_index method
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = check_data_type(self)
+    i = 0
+    until i >= arr.length
+      yield(arr[i], i)
+      i += 1
+    end
+    arr
+  end
+
+  ## my_select method
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = []
+    my_each { |i| arr << i if yield(i) }
+    arr
+  end
+
+  ## my_any method
+  def my_any?
+    unless block_given?
+      for i in self
+        return true if i == true
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return false if i == false
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_any?)
+    end
+    bul = false
+    arr = self
+    arr.my_each { |x| bul = true if yield(x) }
+    bul
+  end
+
+  ## my_all method
+  # rubocop: disable Metrics/CyclomaticComplexity
+  def my_all?
+    unless block_given?
+      for i in self
+        return false if i == false || i.nil?
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return true if i == true
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_all?)
+    end
+    bul = false
+    arr = self
+    arr.my_each { |x| bul = true if yield(x) }
+    bul
+  end
+  # rubocop: enable Metrics/CyclomaticComplexity
+
+  ## my_none method
+  def my_none?
+    unless block_given?
+      for i in self
+        return false if i == true
+      end
+      # rubocop: disable Style/CombinableLoops
+      for i in self
+        return true if i == false
+      end
+      # rubocop: enable Style/CombinableLoops
+      return to_enum(:my_none?)
+    end
+    bul = true
+    arr = self
+    arr.my_each { |x| bul = false if yield(x) }
+    bul
+  end
+
+  ## my_count method
+  def my_count(args = nil)
+    counter = 0
+    unless args.nil?
+      for i in self
+        counter += 1 if i == args
+       end
+      return counter
+    end
+    counter
+  end
```

Comment:
- [ ] #my_count returns the number of items in enum through enumeration
      Failure/Error: expect(array.my_count).to eq array.count.
expected: 100
             got: 0
Refactor the codes to account for this.

- [ ] #my_count returns the number of items in enum through enumeration::range
      Failure/Error: expect(range.my_count).to eq range.count
        expected: 46
             got: 0

- [ ] #my_count counts the number of elements yielding a true value if a block is given
      Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)      
        expected: 41
             got: 0